### DISCUSSION TITLE
Do you have IRC still?

### DISCUSSION THREAD

**Discussion by namgo** [stepmania/stepmania#2220] [Q&A] [answered]
Hi all!

I've got some questions that I think would be best answered informally, which I'll still list here anyways. I'm assuming you left freenode, but maybe this is not the case? I checked libera.chat and didn't find any channels related to stepmania development.

My questions are largely related to what your long-term vision is for stepmania, as there's numerous active forks from various releases of your software. Basically, while SM6 has been scrapped for now, what are your mid-term goals for SM5?

I'm looking for a project (and ideally one where I'm an active user) that I can contribute to. My CV would benefit from such a thing, but I also am looking to develop my skills as a software developer a little bit more formally 😄! I'm not necessarily asking to be mentored through your project, but would certainly appreciate a place to ask about design decisions as I look through some bugs!

Beyond mid-term vision, are you actively looking for help? I'm not specifically looking to contribute to SM5 upstream but am absolutely not opposed to it.

**Comment by geefr** [comment, marked as answer]
Probably the best and most recent summary is here: https://itg.wiki/wiki/StepMania_Forks

There's various discords, per-fork things.

5.1-new is the most up to date branch here, it gets fixes now and then, not aware of an overall plan.

The forks are more active, for their own needs.
(There may be the odd bit of drama back and forth, usually around open vs closed forks)

On a technical side: Stepmania code is a great way to learn patience and debugging. Not such a good place for modern C++ but if you can build and hack this stuff you're on the right track to being an expert :)

**Reply by namgo** [reply]
Thanks for the quick reply! I'm definitely still open to talking to developers further and am less interested in the forks, but I'll start working asynchronously. Any documentation about the layout of the project would still be welcomed (if it doesn't already exist) as I start my journey :)

**Reply by geefr** [reply]
Probably all there is, if you wanted it written down..
https://github.com/stepmania/stepmania/wiki

**Reply by namgo** [reply]
Thanks @geefr ! I've so far read through much of this and it's a good start, my impression is that it's at an awkward place between some development documentation and some user documentation - not even a criticism just something I noticed. I'm open to contributing to bettering documentation but don't know enough about the project to do so yet.

I'm not sure what github etiquette is for opening discussions, let me know if I should open this up for discussion elsewhere.

is a more encompassing test-suite at all useful? I'm under the impression that some issues raised here are due to custom themes and songs being incorrectly written. Your tests don't cover very much right now, and I'm open to expanding this.

On the other hand, there might be other places for me to contribute - I had been active in the Quake World community about a year ago and their engine was so thoroughly to-spec (not to mention ancient) that they felt they didn't really benefit from better tests. I'm open to suggestions for sure!

**Reply by Jaoheah** [reply]
The link is dead. Is there a newer link available?

**Reply by smichel17** [reply]
https://web.archive.org/web/20230107202852/https://itg.wiki/wiki/StepMania_Forks

But just to preserve the information in another place for the future, at this point in time it seems like the main active free software fork is [ITGmania](https://www.itgmania.com/) (https://github.com/itgmania/itgmania/).
